### ISSUE TITLE
Array Physical type `CharacterArraySinglePointer` seems redundant

### ISSUE THREAD

**Issue by Shaikh-Ubaid** [lfortran/lfortran#3967]
With respect to the LLVM Backend, the array physical type `CharacterArraySinglePointer` seems redundant.

For example, in the below it looks like the physical type `CharacterArraySinglePointer` is a combination of `FixedSizeArray` (for character type) and `DescriptorArray` (for character type).

https://github.com/lfortran/lfortran/blob/4628894ed657f18759e5be4f21802a1f323829aa/src/libasr/codegen/llvm_utils.cpp#L1369-L1384

Similarly, one can also see that for computing the Array size, we use the same approaches like a `FixedSizeArray` or a `DescriptorArray` for a `CharacterArraySinglePointer` below

https://github.com/lfortran/lfortran/blob/4628894ed657f18759e5be4f21802a1f323829aa/src/libasr/codegen/asr_to_llvm.cpp#L9713

**Comment by Pranavchiku** [comment]
Nope, we need this for `SciPy`, although work was left incomplete there. I'll link corresponding discussion here.

**Comment by Pranavchiku** [comment]
We need this for
- https://github.com/lfortran/lfortran/issues/2764
- https://github.com/lfortran/lfortran/pull/2810

**Comment by Shaikh-Ubaid** [comment]
Thanks for sharing Pranav. I understand. I think we should reduce the ASR to the smallest possible subset that represents all the features. With that in mind and looking into the `CharacterArraySinglePointer` implementation in the `LLVM` backend, I think we might be able to simply reuse `FixedSizeArray` or `DescriptorArray`.

**Comment by certik** [comment]
I recommend to redesign ASR in this order:

1. Implement proper deallocation
2. Refactor strings to use it and work properly, add all the string physical types that we need
3. Remove any physical types that we do not need (such as `CharacterArraySinglePointer`)
